clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/HubConnection.java

```diff
@@ -143,7 +149,15 @@ public HubConnection(String url, Transport transport, Logger logger, boolean ski
             }
 
             if (negotiateResponse.getAccessToken() != null) {
-                this.headers.put("Authorization", "Bearer " + negotiateResponse.getAccessToken());
+                this.accessTokenProvider = () -> CompletableFuture.completedFuture(negotiateResponse.getAccessToken());
+                String token = "";
+                try {
+                    // We know the future is already completed in this case
+                    // It's fine to call get() on it.
+                    token = this.accessTokenProvider.get().get();
```

Comment:
I'm missing something, the first get is like .invoke?

Reply:
Correct. `accessTokenProvider` is a `Supplier<CompletableFuture<String>>` so you need to call [`Supplier.get()`](https://docs.oracle.com/javase/8/docs/api/java/util/function/Supplier.html#get--) to invoke the method and then [`CompletableFuture.get()`](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html#get--) to resolve the future.

Reply:
Couldn't they have called it invoke,wtf

Reply:
`Supplier.get()`
`Consumer.accept(...)`
`Function.apply(...)`

Reply:
The wording for each of the functional interfaces makes sense based off of the param/return signatures for each. But still a little weird